### PR TITLE
[RHPAM-2684] Add NamespaceDashboard ConsoleLink

### PR DESCRIPTION
**Pull request by ruromero**
Fix https://issues.redhat.com/browse/RHPAM-2684

### CODE REVIEW

**Comment by ruromero** [comment]
/assign @tchughesiv
I still need to test it on different clusters but an initial review would be great. Thanks

**Comment by ruromero** [comment]
ready for review

**Comment by ruromero** [comment]
I don't understand why the `test mod` is failing. I have tried locally with and without VPN by re-cloning the branch and removing the mod cache `rm -rf $GOPATH/pkg/mod/cache/*` and I couldn't reproduce it.
Any hint?

**Comment by akoserwal** [comment]
I deployed on 4.3 cluster, help menu link is showing cluster wide.
Screenshot 2020-02-17 at 1 01 23 PM

**Comment by ruromero** [comment]
> I deployed on 4.3 cluster, help menu link is showing cluster wide.
> Screenshot 2020-02-17 at 1 01 23 PM

@akoserwal the Help Menu Link has been created manually in the cluster by someone else for testing purposes. The one created by this code looks like this:
2020-02-17-085104_1585x478_scrot
In this case, it is a Business Central link but can be Business Central Monitoring or Decision Central, depending on the environment deployed.
Note that it requires an HTTPS link, that's why for the trial environment there shouldn't be a Launcher link.

**Comment by akoserwal** [comment]
@ruromero: got it, thanks for the clarification

**Comment by tchughesiv** [comment]
@ruromero i expect that once this merges and you rebase, the mod check should pass.
https://github.com/kiegroup/kie-cloud-operator/pull/366

File: deploy/catalog_resources/redhat/1.4/businessautomation-operator.1.4.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -185,6 +185,18 @@ spec:
                 name: business-automation-operator
                 resources: {}
               serviceAccountName: business-automation-operator
+      clusterPermissions:
```
Comment (tchughesiv) [inline deploy/catalog_resources/redhat/1.4/businessautomation-operator.1.4.0.clusterserviceversion.yaml]: this new role should be persisted in `csv-gen.go` for future csv generations via `make csv` ... e.g. -
https://github.com/kiegroup/kie-cloud-operator/blob/0c5fb7968959401ddecb50a843fed895c0e4c455/tools/csv-gen/csv-gen.go#L114
https://github.com/kiegroup/kie-cloud-operator/blob/0c5fb7968959401ddecb50a843fed895c0e4c455/tools/csv-gen/csv-gen.go#L298-L299

**Comment by tchughesiv** [comment]
@ruromero seems like the main issue is the CI platform's inability to reach the modules located at `modernc.org` ... which are all new to this PR for some reason. They aren't needed by the main module though...

**Comment by tchughesiv** [comment]
@ruromero these mod replacements seem to solve the connectivity problem...
https://github.com/kiegroup/kie-cloud-operator/pull/370/commits/e7300a0982aee294d5c8bb4770474c12c36cd908
https://github.com/kiegroup/kie-cloud-operator/blob/e7300a0982aee294d5c8bb4770474c12c36cd908/go.mod#L79-L87

the checks are passing here - https://github.com/kiegroup/kie-cloud-operator/pull/370

let me know if you want a PR to your branch...

**Comment by tchughesiv** [comment]
@ruromero can you add a similar section as this for creating the ClusterRole yaml files?
https://github.com/kiegroup/kie-cloud-operator/blob/0c5fb7968959401ddecb50a843fed895c0e4c455/tools/csv-gen/csv-gen.go#L298-L299

**Comment by ruromero** [comment]
> There is way too much going on here! Please separate out the operator SDK upgrade and let's make sure we know what it takes to make that happen... if we can do it right away or not.

Created this PR only with version changes: https://github.com/kiegroup/kie-cloud-operator/pull/386
Once the linked PR is merged I will rebase this one including only ConsoleLink related changes.

**Comment by ruromero** [comment]
@bmozaffa I have rebased from #386 and added commit https://github.com/kiegroup/kie-cloud-operator/pull/365/commits/e7d5c5787c5b3f2ce9518dc8d4c979d775ed491b

**Comment by tchughesiv** [comment]
please rebase

**Comment by ruromero** [comment]
rebased

File: deploy/cluster_role_binding.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: kie-cloud-operator
+roleRef:
+  kind: ClusterRole
+  name: kie-cloud-operator
```
[the remark this answers is not in the document]
Comment (ruromero) [reply deploy/cluster_role_binding.yaml]: Added, thanks!

**Comment by bmozaffa** [comment]
I think we need to roll this back. While I think finalizers are a great addition to operator-utils, I don't know why reconciliation for console link should be done any differently than any other resource reconciliation we do in this operator, namely:

Based on input, in this case CR status, figure out what should be created. Use operator-utils/resource libs to create/update/delete as needed. @ruromero , any reason why that wouldn't work here?

**Pull request merged**